### PR TITLE
Grafana: Adds support for PKCS1 encrypted certs

### PR DESCRIPTION
**Pull request by leandro-deveikis**
**What is this feature?**

Adds support for PKCS#1 (RSA) certificates encrypted with password.

**Why do we need this feature?**

Fixes https://github.com/grafana/grafana-operator-experience-squad/issues/968

### CODE REVIEW

File: pkg/api/http_server.go [hunk 1]
```diff
@@ -868,6 +868,20 @@ func handleEncryptedCertificates(cfg *setting.Cfg) (*tls.Certificate, error) {
 		if err != nil {
 			return nil, fmt.Errorf("error marshaling PKCS8 Private key: %w", err)
 		}
+	} else if strings.Contains(keyPemBlock.Type, "RSA PRIVATE KEY") {
+		// Check if the PEM block is encrypted with PKCS#1
+		// This is still requested by some customers (and fairly used)
+		// nolint:staticcheck
+		if x509.IsEncryptedPEMBlock(keyPemBlock) {
```
Comment (DanCech) [inline pkg/api/http_server.go:875]: suggest inverting this test, so the happy path continues down and the error path goes to the right

**Review by DanCech — APPROVED** [review]
Looks good, one minor readability suggestion

File: pkg/api/http_server.go (hunk 1 above)
Comment (leandro-deveikis) [reply pkg/api/http_server.go]: fixed in 74d241a9ea120b3acee9f67d621a5d6f350703d5
thanks!

**Review by macabu — APPROVED** [review]

**Review by mmandrus — APPROVED** [review]
LGTM!

**Pull request merged**